fetchai/ledger/transaction.py

```diff
@@ -146,4 +206,54 @@ def synergetic_data_submission(self, is_submission):
 
     def add_signer(self, signer: Identity):
         if signer not in self._signers:
-            self._signers[signer] = None  # will be replaced with a signature in the future
+            self._signers[signer] = {}  # will be replaced with a signature in the future
+
+    def sign(self, signer: Entity):
+        if Identity(signer) in self._signers:
+            signature = signer.sign(self.payload)
+            self._signers[Identity(signer)] = {
+                'signature': signature,
+                'verified': signer.verify(self.payload, signature)
+            }
+
+    def merge_signatures(self, tx2: 'Transaction'):
```

Comment:
This method returns void usually but on error it returns none and never throws. This is different to pattern elsewhere in python sdk maybe refactor.